ygot/render.go

```diff
@@ -536,26 +744,17 @@ func constructJSON(s GoStruct, parentMod string, args jsonOutputConfig) (map[str
 					continue
 				}
 			case 1:
-				pelem, ok := p[0].(string)
-				if !ok {
-					errs.Add(fmt.Errorf("could not convert path %v into a string", p))
-					continue
-				}
+				pelem := p.stringSlicePath[0]
```

Comment:
p.At(0), here and below.

Reply:
Per discussion, let's make this StringAt() (string, error) and PathElemAt() etc.
You can revisit further abstraction if it makes sense in the future.

Reply:
Done.